src/base/net/downloadhandler.cpp

```diff
@@ -155,13 +155,17 @@ void Net::DownloadHandler::checkDownloadSize(qint64 bytesReceived, qint64 bytesT
     }
 }
 
-void Net::DownloadHandler::handleRedirection(QUrl newUrl)
+void Net::DownloadHandler::handleRedirection(const QUrl &newUrl)
 {
-    // Resolve relative urls
-    if (newUrl.isRelative())
-        newUrl = m_reply->url().resolved(newUrl);
+    if (m_downloadRequest.redirectionCounter() >= MAX_REDIRECTIONS) {
+        emit downloadFailed(url(), tr("Exceeded max redirections (%1)").arg(MAX_REDIRECTIONS));
+        this->deleteLater();
+        return;
+    }
 
-    const QString newUrlString = newUrl.toString();
+    // Resolve relative urls
+    const QUrl resolvedUrl = (newUrl.isRelative()) ? m_reply->url().resolved(newUrl) : newUrl;
```

Comment:
Still, it looks like a "witch hunt". Do you want to destroy all the variables at the root?
I really can't figure it out... Why do we need to generate several immutable things, if you can use only one mutable?

Reply:
I reverted it, yet haven't pushed to github yet.

Isn't immutable a good property?
Honestly I cannot understand your concern.

Reply:
>I reverted it, yet haven't pushed to github yet.

Sorry. Please don't revert it. It wasn't my intention.
Besides, I messed up something, so this case is not the right place for my complaints. It's ok since you changed parameter to const ref.